src/client/actions/MessageReactionAdd.js

```diff
@@ -9,16 +9,30 @@ const Constants = require('../../util/Constants');
 */
 
 class MessageReactionAdd extends Action {
-  handle(data) {
+  async handle(data) {
     const user = this.client.users.get(data.user_id);
     if (!user) return false;
     // Verify channel
     const channel = this.client.channels.get(data.channel_id);
     if (!channel || channel.type === 'voice') return false;
     // Verify message
     const message = channel.messages.get(data.message_id);
-    if (!message) return false;
     if (!data.emoji) return false;
+    if (!message) {
+      if (this.client.options.autofetch.includes(Constants.WSEvents.MESSAGE_REACTION_ADD)) {
+        // Check if message is fetchable
+        if (!channel.permissionsFor(channel.guild.me).has('READ_MESSAGES')) return false;
+
+        try {
+          const fetchedMessage = await channel.fetchMessage(data.message_id);
+          const reaction = fetchedMessage._addReaction(data.emoji, user);
+          this.client.emit(Constants.Events.MESSAGE_REACTION_ADD, reaction, user);
```

Comment:
Shouldn't you return the message, the reaction and the user here like line 40 does it too?